### ISSUE TITLE
Orders section 50 items limitation and missed older orders log

### ISSUE THREAD

**Issue by UmckaBear** [BuildOnViction/bug-reports#33]
The sum of  Open Orders+Order History is always 50. For example if you have 1 closed order  in Order History then you will see only 49 Open orders. If you have 5 closed orders and 45 opened, and want add more orders, older orders will be removed from the list. You can see them in blockchain (if you saved transaction), but can't see and cancel on exchange. Order still in order list.

Video example:
https://drive.google.com/file/d/1ARPJ393-Ujbs_1P6fF_4SFZ-oxUJawKX/view?usp=sharing

When you fill the Order History with 50 orders, then after opening new order, the oldest from Order History will be removed to balance 50 count. In such way with adding new open orders you can clear Order History.

![Order History](https://user-images.githubusercontent.com/31537952/72473712-f6781e00-37ef-11ea-8fd9-dee0606e6a5f.jpg)

In video example 2 orders were removed from the list (22:39:53 and 22:39:57 creation time) after adding few new.

https://drive.google.com/file/d/1pxhdMhKw0AbCDNdfJataFXps-a3S7-JE/view?usp=sharing

This order from 22:39:53 is still opened but you can't check it in Open orders list.
https://scan.testnet.tomochain.com/orders/0x7e8403ae0863c9d02e326914d1be503db24906c95ea252df8a70d97925a18212

On screen below you can see only one order for 6 TOMO, but in orders more then 1k TOMO.
![Missed orders](https://user-images.githubusercontent.com/31537952/72472925-43f38b80-37ee-11ea-96bb-f61b3a626fd0.jpg)

**Comment by UmckaBear** [comment]
I checked with normal idles between opening/closing orders. Checked without idles (only message confirmation received). You can use rejected orders, closed, cancelled orders = there will be same result with 50 limit after page refreshing.

The best way to empty Open Orders list is to use Rejected orders. In this case will be created new open order and then closed as rejected.

**Comment by thanhson1085** [comment]
Now, Open Orders limits to show 50 orders.
So you expect to see more, right?

**Comment by UmckaBear** [comment]
> Now, Open Orders limits to show 50 orders.
> So you expect to see more, right?

Probably, Open Orders  was intended to show only the last 50 open orders, same as Order History. And in future you will add tab to show all opened orders/history. What we have right now:
1. The bug, that you can't see 50 open orders and closed 50 orders (cancelled, rejected, filled) at the same time after page refresh/opening other pair. It looks like two glasses with a volume of water for one glass. 45-5, 50-0, 1-49. Queue, where all orders (opened, canceled) in one line. Last comes - older out of vision.
2. Older orders can be removed due this limitations and you will not be able check them later. Best way use Rejected order as in video below. At the end of the video I have 46 opened order, created 3 more (they were rejected) and count of open orders decreased by 3 to 43 **after page refresh as shown**, since 3 orders were added to Orders history.
https://drive.google.com/file/d/1JsE_PQP5jItkdjNdUDLjY0dCBbI310Px/view?usp=sharing

**Comment by thanhson1085** [comment]
> The bug, that you can't see 50 open orders and closed 50 orders (cancelled, rejected, filled) at the same time after page refresh/opening other pair. It looks like two glasses with a volume of water for one glass. 45-5, 50-0, 1-49. Queue, where all orders (opened, canceled) in one line. Last comes - older out of vision.

Did you mean, if you have more than 50 open orders, after refreshing the page, the open order table does not show it correctly (e.g, show only 46 orders), right?

> Older orders can be removed due this limitations and you will not be able check them later. Best way use Rejected order as in video below. At the end of the video I have 46 opened order, created 3 more (they were rejected) and count of open orders decreased by 3 to 43 after page refresh as shown, since 3 orders were added to Orders history.
> https://drive.google.com/file/d/1JsE_PQP5jItkdjNdUDLjY0dCBbI310Px/view?usp=sharing

The the video, you created the orders too small amount (> 1 TOMO), it will be rejected, because it doesn't have enough TOMO to pay the trade fee for masternode.

@UmckaBear , In order to save our time, you should create the issue with the short sentences, and it should contain 2 parts:
- issue description (text, videos, image)
- your expectation

**Comment by UmckaBear** [comment]
> Did you mean, if you have more than 50 open orders, after refreshing the page, the open order table does not show it correctly (e.g, show only 46 orders), right?

Yes. If you have, for example, 4 closed orders and 50 open orders, after refresh you will see 46 open orders and 4 closed orders. 

> The the video, you created the orders too small amount (> 1 TOMO), it will be rejected, because it doesn't have enough TOMO to pay the trade fee for masternode.

You didn't get the idea of showed video. Rejected order used to create new order and then move it into Order History as Rejected. This is the quickest way place item in both sections and show how it affect displayed items. 

> @UmckaBear , In order to save our time, you should create the issue with the short sentences, and it should contain 2 parts:
> 
> * issue description (text, videos, image)
> * your expectation

1. Issue description:
Open Orders and Order History sections have incorrect count for placed items. They have a common counter for showed items. Max displayed Open Orders + max Order History = 50 items always. After reaching the maximum value of 50 and further work with orders you will get incorrect display of items in both sections.

In video below I have reached max 50 items. Then I added new 4 orders, which were Rejected and placed to the Order History. There should be 50 open orders and 4 rejected. But after refresh there 46-4. Then I've added 3 more open orders, which were Rejected. As result 43-7 instead of 50-7. 
https://drive.google.com/file/d/1JsE_PQP5jItkdjNdUDLjY0dCBbI310Px/view?usp=sharing


2. Expectation:
If you would like to use 50 items limitation, then Open Orders should correctly display last 50 items in case of big open orders count (more then 50). And Order History should also correctly display max items count. In other words, at the same time you can have correctly displayed up to 50 items in both sections.
Adding items to one section should not affect displayed items in other one after refresh.


@thanhson1085 It is difficult to explain in few words some bugs, since there should be reached some conditions. In our case it is max showed items. The bug is not at all obvious. I discovered it when I saw the loss of about 30 open orders and began to dig deeper. That's why I'm trying to give detailed description.

**Comment by thanhson1085** [comment]
Your report requires much time to reproduce. We will do it when we have time, thank you.

**Comment by UmckaBear** [comment]
> Your report requires much time to reproduce. We will do it when we have time, thank you.

No problem. You took note of it and that's most importantly.

As I noted previously, this bug is not at all obvious, I saw this issue only in active exchange using. Probably users, who made 1-2 orders per day, never have paid attention on it. But active user with big activity will run into problems sooner or later.
